snuba/util.py

```diff
@@ -207,24 +207,24 @@ def is_condition(cond_or_list):
     )
 
 
-def all_referenced_columns(body):
+def all_referenced_columns(query):
     """
     Return the set of all columns that are used by a query.
     """
     col_exprs = []
 
     # These fields can reference column names
     for field in ['arrayjoin', 'groupby', 'orderby', 'selected_columns']:
-        if field in body:
-            col_exprs.extend(to_list(body[field]))
+        if field in query:
+            col_exprs.extend(to_list(query[field]))
 
     # Conditions need flattening as they can be nested as AND/OR
-    if 'conditions' in body:
-        flat_conditions = list(chain(*[[c] if is_condition(c) else c for c in body['conditions']]))
+    if 'conditions' in query:
+        flat_conditions = list(chain(*[[c] if is_condition(c) else c for c in query['conditions']]))
         col_exprs.extend([c[0] for c in flat_conditions])
 
-    if 'aggregations' in body:
-        col_exprs.extend([a[1] for a in body['aggregations']])
+    if 'aggregations' in query:
+        col_exprs.extend([a[1] for a in query['aggregations']])
 
     # Return the set of all columns referenced in any expression
     return set(chain(*[columns_in_expr(ex) for ex in col_exprs]))
```

Comment:
This function is only used by places that can be restricted to the `query` structure and doesn't require any of the `options` data.